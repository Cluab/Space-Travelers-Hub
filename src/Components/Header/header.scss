.header {
  padding: 1rem 2rem;
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: #fff;
  border-bottom: 1px solid #cacaca;

  & > nav {
    display: flex;
    gap: 2.5rem;
    width: 100%;
    justify-content: space-between;
  }

  & > nav > ul {
    list-style: none;
    display: flex;
    gap: 2rem;

    & a {
      text-decoration-line: none;
      line-height: 25px;
      color: #449cff;
      font-family: Poppins, sans-serif;
      font-weight: 500;
    }

    & > li:last-child {
      position: relative;

      &::after {
        content: " ";
        position: absolute;
        width: 2px;
        height: 18px;
        left: -18px;
        top: 3px;
        border-radius: 8px;
        background-color: #acacac;
      }
    }
  }

  & > nav h1 {
    margin: 0;
    display: flex;
    align-items: center;
    font-family: Poppins, sans-serif;
    gap: 1rem;

    & > a {
      text-decoration-line: none;
      color: #494949;
    }
  }
}
