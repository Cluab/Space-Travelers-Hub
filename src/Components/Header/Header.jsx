import { NavLink } from 'react-router-dom';
import SpaceHubLogo from '../../Assets/space-hub-logo.png';
import './header.scss';

const Header = () => {
  const activeStyle = {
    textDecorationLine: 'underline',
  };

  return (
    <header className="header">
      <nav>
        <h1>
          <img src={SpaceHubLogo} width="50" alt="Space travels' hub logo" />
          <NavLink to="/">Space Travels&apos; Hub</NavLink>
        </h1>
        <ul>
          <li>
            <NavLink
              to="/"
              style={({ isActive }) => (isActive ? activeStyle : undefined)}
            >
              Rockets
            </NavLink>
          </li>
          <li>
            <NavLink
              to="missions"
              style={({ isActive }) => (isActive ? activeStyle : undefined)}
            >
              Missions
            </NavLink>
          </li>
          <li>
            <NavLink
              to="profile"
              style={({ isActive }) => (isActive ? activeStyle : undefined)}
            >
              My Profile
            </NavLink>
          </li>
        </ul>
      </nav>
    </header>
  );
};

export default Header;
