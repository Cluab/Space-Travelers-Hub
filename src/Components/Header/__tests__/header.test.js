import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Header from '../Header';
import '@testing-library/jest-dom/extend-expect';

describe('nav bar test cases', () => {
  render(<BrowserRouter><Header /></BrowserRouter>);

  test('test for logo visabelaty', () => {
    const pageLogo = screen.getByAltText("Space travels' hub logo");
    expect(pageLogo).toBeVisible();
  });

  it('test nav links to have an under line to them', () => {
    const linkButton = screen.getByText('Missions');

    fireEvent.click(linkButton);

    expect(linkButton).toHaveClass('activeStyle');
  });
});
