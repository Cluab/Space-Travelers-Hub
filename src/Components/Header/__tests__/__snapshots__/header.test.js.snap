// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test cases should  1`] = `
<header
  className="header"
>
  <nav>
    <h1>
      <img
        alt="Space travels' hub logo"
        src="space-hub-logo.png"
        width="50"
      />
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        Space Travels' Hub
      </a>
    </h1>
    <ul>
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "textDecorationLine": "underline",
            }
          }
        >
          Rockets
        </a>
      </li>
      <li>
        <a
          className=""
          href="/missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li>
        <a
          className=""
          href="/profile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
